### PR TITLE
Enable rest java chart by default

### LINKED ISSUE #8105: Enable rest-java chart by default
### Problem

The rest-java pods do not currently run in production.

### Solution

* Add nft allowances to ingress route
* Change rest-java chart to be enabled by default
* Increase request resources in values.yaml
* Decrease request resources in ci
* Unset override in integration flux config

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jascks**
**Description**:
Ensure the rest-java pods run by default in all prod and non-prod environments.

This PR:
* Add nft allowances to ingress route
* Change rest-java chart to be enabled by default
* Increase request resources in values.yaml
* Decrease request resources in ci
* Unset override in integration flux config (PR #8118).
* Remove placeholder DummyController and related ingress route

**Related issue(s)**:

Fixes #8105

**Notes for reviewer**:
This PR depends on PR #8120 to resolve the same underlying issue.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by steven-sheehy** [comment]
CI is failing since 0.0.2 doesn't exist in demo bucket. Need to see if there's any row in entity table for demo bucket and customize with that entity in CI.

File: charts/hedera-mirror-rest-java/postman.json [hunk 1]
```diff
@@ -20,25 +20,29 @@
 									"    pm.response.to.have.status(200);",
 									"});",
 									"pm.test(\"Body matches string\", function () {",
-									"    pm.expect(pm.response.text()).to.include(\"Hello world\");",
+									"    pm.expect(pm.response.text()).to.include(\"\\\"allowances\\\":[]\");",
```
Comment (xin-hedera) [inline charts/hedera-mirror-rest-java/postman.json:23]: nit, it's better to do exact match of the JSON response

```
{
  "allowances": [],
  "links": {
    "next": null
  }
}
```

File: charts/hedera-mirror/templates/NOTES.txt [hunk 2]
```diff
@@ -35,11 +35,11 @@ To access the REST API:
...
 To access the Java REST API:
 {{- if contains "LoadBalancer" .Values.restjava.service.type }}
-  REST_IP="$(kubectl get "service/{{ .Release.Name }}-rest-java" -n {{ include "hedera-mirror.namespace" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+  REST_IP="$(kubectl get "service/{{ .Release.Name }}-restjava" -n {{ include "hedera-mirror.namespace" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')"
 {{- else if contains "ClusterIP" .Values.restjava.service.type }}
-  kubectl port-forward "service/{{ .Release.Name }}-rest-java" -n {{ include "hedera-mirror.namespace" . }} {{ .Values.rest.service.port }}:80 &
+  kubectl port-forward "service/{{ .Release.Name }}-restjava" -n {{ include "hedera-mirror.namespace" . }} {{ .Values.rest.service.port }}:80 &
 {{- end }}
-  curl -s "http://${REST_IP:-127.0.0.1}:{{.Values.restjava.service.port}}/api/v1/dummy"
+  curl -s "http://${REST_IP:-127.0.0.1}:{{.Values.restjava.service.port}}/api/v1/accounts/0.0.2/allowances/nfts"
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply charts/hedera-mirror/templates/NOTES.txt:42]: Except other chart users can customize it so we should keep it using the helm property

File: charts/hedera-mirror/templates/NOTES.txt [hunk 3]
```diff
@@ -35,11 +35,11 @@ To access the REST API:
 {{- if and .Values.restjava.enabled .Values.restjava.service }}
 To access the Java REST API:
 {{- if contains "LoadBalancer" .Values.restjava.service.type }}
-  REST_IP="$(kubectl get "service/{{ .Release.Name }}-rest-java" -n {{ include "hedera-mirror.namespace" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+  REST_IP="$(kubectl get "service/{{ .Release.Name }}-restjava" -n {{ include "hedera-mirror.namespace" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')"
 {{- else if contains "ClusterIP" .Values.restjava.service.type }}
-  kubectl port-forward "service/{{ .Release.Name }}-rest-java" -n {{ include "hedera-mirror.namespace" . }} {{ .Values.rest.service.port }}:80 &
+  kubectl port-forward "service/{{ .Release.Name }}-restjava" -n {{ include "hedera-mirror.namespace" . }} {{ .Values.rest.service.port }}:80 &
```
Comment (steven-sheehy) [inline charts/hedera-mirror/templates/NOTES.txt:40]: There's a copy/paste error here from when this was initially added. Also, we can utilize the named `http` port as using port 80 will break if we change it. Finally, service.port is port 80 which will fail to bind without sudo. Probably we should use a hardcoded port >= 1024 like 8084. Then we can use Xin's previous suggestion.

```suggestion
  kubectl port-forward "service/{{ .Release.Name }}-restjava" -n {{ include "hedera-mirror.namespace" . }} 8084:http &
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/hedera-mirror/templates/NOTES.txt (hunk 3 above)
Comment (jascks) [reply charts/hedera-mirror/templates/NOTES.txt:40]: That make sense. I changed it manually myself to 9999 when testing it out. 😄 

Questions:

1. The other services (except grpc) forward to local port 80 as well, presenting the same problem. Should I change the all?
2. Only one can bind to 8084 at a time, should each service bind to a unique local port, say starting at 8080 or someothing?

File: charts/hedera-mirror-rest-java/postman.json (hunk 1 above)
Comment (jascks) [reply charts/hedera-mirror-rest-java/postman.json:23]: Done. Updated in postman and exported to this file.

Screenshot 2024-04-19 at 8 46 23 AM

File: charts/hedera-mirror/templates/NOTES.txt (hunk 3 above)
Comment (steven-sheehy) [reply charts/hedera-mirror/templates/NOTES.txt:40]: You can change them all as long as you test each out by running `helm template` and running the rendered commands. Each app should just use its internal port that it actually listens on, e.g. importer uses 8080, etc.

Comment (jascks) [reply charts/hedera-mirror/templates/NOTES.txt]: Okay, I'm taking a more minimalist approach and sticking to the original suggestions.

Deploying to minikube, this section renders as:
```
To access the Java REST API:
  kubectl port-forward "service/mirror-restjava" -n mirror-jeff 8084:80 &
  curl -s "http://${REST_IP:-127.0.0.1}:8084/api/v1/accounts/0.0.2/allowances/nfts"
```

The port forward works.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**